### PR TITLE
support set value of tidb_opt_distinct_agg_push_down in configure file

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Problem Summary:
TiDB has a variable named `tidb_opt_distinct_agg_push_down`, agg with distinct will be pushdown to cop if this variable is set to true, however, this is only a session level configuration, some customers complain that it is inconvenient to use. This pr support set the value of `tidb_opt_distinct_agg_push_down` in configure file, so users do not have the set it in every connection.

### What is changed and how it works?

make `tidb_opt_distinct_agg_push_down` an instance level variable

Tests 
- Manual test (add detailed scripts or steps below)
1. add distinct-agg-push-down = true in config file, restart TiDB, select @@tidb_opt_distinct_agg_push_down returns 1
2. add distinct-agg-push-down = false in config file, restart TiDB, select @@tidb_opt_distinct_agg_push_down returns 0

### CODE REVIEW

**Comment by windtalker** [comment]
> Better to add this config in config.toml.example

Done

File: config/config.toml.example [hunk 1]
```diff
@@ -232,6 +232,9 @@ force-priority = "NO_PRIORITY"
 # Bind info lease duration, which influences the duration of loading bind info and handling invalid bind.
 bind-info-lease = "3s"
 
+# Whether support push down aggregation with distinct to cop task
```
Comment (SunRunAway) [inline config/config.toml.example:235]:
```suggestion
# Whether support pushing down aggregation with distinct to cop task
```

**Comment by zz-jason** [comment]
@windtalker do we need to cherry-pick this PR to release branches?

**Comment by zz-jason** [comment]
@windtalker IMO, this PR needs a release note, please write a release note in the PR description.

**Comment by windtalker** [comment]
> @windtalker do we need to cherry-pick this PR to release branches?

Yes, this should be cherry-picked to 4.0 branch

**Pull request merged**